staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -564,13 +564,17 @@ func (f *featureGate) featureSpec(key Feature) (FeatureSpec, error) {
 }
 
 func (f *featureGate) unsafeRecordQueried(key Feature) {
-	queriedFeatures := map[Feature]struct{}{}
-	for k := range f.queriedFeatures.Load().(map[Feature]struct{}) {
-		queriedFeatures[k] = struct{}{}
+	queriedFeatures := f.queriedFeatures.Load().(map[Feature]struct{})
+	if queriedFeatures == nil {
+		queriedFeatures = map[Feature]struct{}{}
 	}
 	if _, ok := queriedFeatures[key]; ok {
 		return
 	}
+	queriedFeatures = map[Feature]struct{}{}
+	for k := range f.queriedFeatures.Load().(map[Feature]struct{}) {
```

Comment:
The map should only be loaded one time. The second load could return a completely different map, and we've only checked that the key is missing from the first map.